libs/api/domains/cms/src/lib/models/projectPage.model.ts

```diff
@@ -0,0 +1,67 @@
+import { Field, ObjectType, ID } from '@nestjs/graphql'
+
+import { IProjectPage } from '../generated/contentfulTypes'
+import {
+  mapDocument,
+  safelyMapSliceUnion,
+  SliceUnion,
+} from '../unions/slice.union'
+import { GenericTag, mapGenericTag } from './genericTag.model'
+import { Link, mapLink } from './link.model'
+import { mapProjectSubpage, ProjectSubpage } from './projectSubpage.model'
+
+@ObjectType()
+export class ProjectPage {
+  @Field(() => ID)
+  id!: string
+
+  @Field()
+  title!: string
+
+  @Field()
+  slug!: string
+
+  @Field()
+  theme!: string
+
+  @Field()
+  sidebar!: boolean
+
+  @Field(() => [Link])
+  sidebarLinks!: Array<Link>
+
+  @Field()
+  subtitle!: string
+
+  @Field()
+  intro!: string
+
+  @Field(() => [SliceUnion], { nullable: true })
+  content?: Array<typeof SliceUnion>
+
+  @Field(() => [SliceUnion])
+  slices!: Array<typeof SliceUnion | null>
+
+  @Field(() => GenericTag, { nullable: true })
+  newsTag!: GenericTag | null
+
+  @Field(() => [ProjectSubpage])
+  projectSubpages!: Array<ProjectSubpage>
+}
+
+export const mapProjectPage = ({ sys, fields }: IProjectPage): ProjectPage => ({
+  id: sys.id,
+  title: fields.title ?? '',
```

Comment:
are these fields ever undefined/null? if this is mostly to appease the type system, you should be able to get away with doing `xxx.yy as string` instead of assigning a default

Reply:
This is how all the other CMS models do it, I think it's best to follow the trend :stuck_out_tongue: 

When the Contentful preview API is used (for example on the beta), all fields can be null. The defaults are provided so that WIP content can be previewed.